activerecord-sqlserver-adapter.gemspec

```diff
@@ -17,5 +17,4 @@ Gem::Specification.new do |spec|
   spec.test_files    = spec.files.grep(%r{^(test|spec|features)/})
   spec.require_paths = ['lib']
   spec.add_dependency 'activerecord', '~> 5.1.0'
-  spec.add_dependency 'tiny_tds'
```

Comment:
And fwiw a gemspec is just Ruby code so we can platform detect and exclude for JRuby specifically.

Reply:
@enebo you can't just conditionally change the gemspec via standard Ruby and expect it will work. The gemspec is serialized to `yaml` at gem build time and stored in the gem metadata, so you must make sure the gemspec is accurate build time so that the correct serialized version is in the resulting gem.

See the `ffi` gem for an example of one way to do it.

Reply:
@coderjoe weirdly I was not aware of the yaml.  Whenever I profile rubygems startup time I see zillions of .gemspecs being eval'd.  I can imagine rubygems.org and other tools do want this to just be static data at publish time though.

So the problem as described about proper serialization would be the concern if this was run from JRuby to install this gem it would serialize a YAML file without tiny_tds in it?  I was just thinking the second run to build the jruby java bits could load the same file for it's gemspec vs having a second file (or embedded spec in Rakefile) while it made it's native parts.  Since I did not know about the YAML serialization into the .gem file I may be way off base.

Reply:
I should add that if the answer to my question was yes I could see that being enough of a concern to not do it.  I am most just making sure I understand.

Reply:
Yeah I that's the general concern. It would mean the output of the task to create the gem would depend entirely on whether or not the user was using `jruby` or `MRI`. I'd much prefer to have a separate rake task to create both from one sitting.

For reference in FFI: https://github.com/ffi/ffi/blob/0ee4f7931bb56c7d0e11ef66c51711cb59a80bc4/Rakefile#L154

Reply:
>  I'd much prefer to have a separate rake task to create both from one sitting.

isn't best, Bundler needs a gemspec + if by any chance you end up having some native parts.
... in which case you need to built under the target Ruby.
its fairly common (e.g. nokogiri) and can also can get it auto-building with `gem '', github: ...`
should stress out the 'some' native code popping up, otherwise doesn't matter much. 
on the other hand, if you do not need to be building an ext why exclude `gem '', github: ...` from working with Bundler (having a proper gemspec behaving under the target Ruby).